task-sdk/tests/task_sdk/definitions/test_secrets_masker.py

```diff
@@ -729,3 +731,356 @@ def test_mixed_structured_unstructured_data(self):
             assert redacted_data["nested"]["info"] == "No secrets here"
+
+
+class TestSecretsMaskerMerge:
+    """Test the merge functionality for restoring original values from redacted data."""
+
+    @pytest.mark.parametrize(
+        ("new_value", "old_value", "name", "expected"),
+        [
+            ("***", "original_secret", "password", "original_secret"),
+            ("new_secret", "original_secret", "password", "new_secret"),
+            ("***", "original_value", "normal_field", "***"),
+            ("new_value", "original_value", "normal_field", "new_value"),
+            ("***", "original_value", None, "***"),
+            ("new_value", "original_value", None, "new_value"),
+        ],
+    )
+    def test_merge_simple_strings(self, new_value, old_value, name, expected):
+        secrets_masker = SecretsMasker()
+        with patch("airflow.sdk.execution_time.secrets_masker._secrets_masker", return_value=secrets_masker):
+            result = merge(new_value, old_value, name)
+            assert result == expected
+
+    def test_merge_dictionaries(self):
+        secrets_masker = SecretsMasker()
+
+        old_data = {
+            "password": "original_password",
+            "api_key": "original_api_key",
+            "normal_field": "original_normal",
+            "token": "original_token",
+        }
+
+        new_data = {
+            "password": "***",
+            "api_key": "new_api_key",
+            "normal_field": "new_normal",
+            "token": "***",
+        }
+
+        expected = {
+            "password": "original_password",
+            "api_key": "new_api_key",
+            "normal_field": "new_normal",
+            "token": "original_token",
+        }
+
+        with patch("airflow.sdk.execution_time.secrets_masker._secrets_masker", return_value=secrets_masker):
+            result = merge(new_data, old_data)
+            assert result == expected
+
+    def test_merge_nested_dictionaries(self):
+        secrets_masker = SecretsMasker()
+
+        old_data = {
+            "config": {"password": "original_password", "host": "original_host"},
+            "credentials": {"api_key": "original_api_key", "username": "original_user"},
+        }
+
+        new_data = {
+            "config": {
+                "password": "***",
+                "host": "new_host",
+            },
+            "credentials": {
+                "api_key": "new_api_key",
+                "username": "new_user",
+            },
+        }
+
+        expected = {
+            "config": {
+                "password": "original_password",
+                "host": "new_host",
+            },
+            "credentials": {
+                "api_key": "new_api_key",
+                "username": "new_user",
+            },
+        }
+
+        with patch("airflow.sdk.execution_time.secrets_masker._secrets_masker", return_value=secrets_masker):
+            result = merge(new_data, old_data)
+            assert result == expected
```

Comment:
Oh i see you are covering it here? Maybe merge the two tests into one with pytest params?

Reply:
Done